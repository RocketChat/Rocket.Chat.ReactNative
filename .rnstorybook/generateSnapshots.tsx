import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
