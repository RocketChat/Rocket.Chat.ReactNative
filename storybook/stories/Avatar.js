import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { themes } from '../../app/constants/colors';
import Avatar from '../../app/containers/Avatar/Avatar';
import Status from '../../app/containers/Status/Status';
import StoriesSeparator from './StoriesSeparator';
import sharedStyles from '../../app/views/Styles';

const styles = StyleSheet.create({
	status: {
		borderWidth: 4,
		bottom: -4,
		right: -4
	},
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

const Separator = ({ title, theme }) => <StoriesSeparator title={title} theme={theme} />;
Separator.propTypes = {
	title: PropTypes.string,
	theme: PropTypes.string
};

const AvatarStories = ({ theme }) => (
	<ScrollView style={{ backgroundColor: themes[theme].backgroundColor }}>
		<Separator title='Avatar by text' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by roomId' theme={theme} />
		<Avatar
			type='p'
			rid='devWBbYr7inwupPqK'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by url' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by path' theme={theme} />
		<Avatar
			avatar='/avatar/diego.mello'
			server={server}
			size={56}
		/>
		<Separator title='With ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			avatarETag='5ag8KffJcZj9m5rCv'
			server={server}
			size={56}
		/>
		<Separator title='Without ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			server={server}
			size={56}
		/>
		<Separator title='Emoji' theme={theme} />
		<Avatar
			emoji='troll'
			getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })}
			server={server}
			size={56}
		/>
		<Separator title='Direct' theme={theme} />
		<Avatar
			text='diego.mello'
			server={server}
			type='d'
			size={56}
		/>
		<Separator title='Channel' theme={theme} />
		<Avatar
			text='general'
			server={server}
			type='c'
			size={56}
		/>
		<Separator title='Touchable' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			onPress={() => console.log('Pressed!')}
			size={56}
		/>
		<Separator title='Static' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			isStatic
			size={56}
		/>
		<Separator title='Custom borderRadius' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			borderRadius={28}
			size={56}
		/>
		<Separator title='Children' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		>
			<Status
				size={24}
				style={[sharedStyles.status, styles.status]}
				theme={theme}
			/>
		</Avatar>
		<Separator title='Wrong server' theme={theme} />
		<Avatar
			text='Avatar'
			server='https://google.com'
			size={56}
		/>
		<Separator title='Custom style' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
			style={styles.custom}
		/>
	</ScrollView>
);
AvatarStories.propTypes = {
	theme: PropTypes.string
};
export default AvatarStories;
