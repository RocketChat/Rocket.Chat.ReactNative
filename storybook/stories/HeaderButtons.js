/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions, react/prop-types */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import * as HeaderButton from '../../app/containers/HeaderButton';
import Header from '../../app/containers/Header';
import { ThemeContext } from '../../app/theme';

const stories = storiesOf('Header Buttons', module);

const HeaderExample = ({ left, right }) => (
	<Header
		headerLeft={left}
		headerTitle={() => <View style={{ flex: 1 }} />}
		headerRight={right}
	/>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }) => (
	<ThemeContext.Provider
		value={{ theme }}
	>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Drawer />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CloseModal />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CancelModal />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.More />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Download />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Preferences />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Legal />
			)}
		/>
	</>
));
