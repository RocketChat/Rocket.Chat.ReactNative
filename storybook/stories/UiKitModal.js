import React from 'react';
import { ScrollView, StyleSheet, SafeAreaView } from 'react-native';

import { UiKitModal, UiKitComponent } from '../../app/containers/UIKit';
import { KitContext, defaultContext } from '../../app/containers/UIKit/utils';
import StoriesSeparator from './StoriesSeparator';

// eslint-disable-next-line react/prop-types
const Separator = ({ title }) => <StoriesSeparator title={title} theme='light' />;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff'
	},
	padding: {
		paddingHorizontal: 16
	}
});

export default () => (
	<SafeAreaView style={styles.container}>
		<ScrollView style={[styles.container, styles.padding]} keyboardShouldPersistTaps='always'>
			<Separator title='Modal - Section and Selects' />
			{
				UiKitModal([
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Rocket.Chat is free, unlimited and open source* ðŸš€\nIf you have any doubt ask to @rocketcat'
						}
					},
					{
						type: 'divider'
					},
					{
						type: 'section',
						fields: [
							{
								type: 'mrkdwn',
								text: '*Text 1*\nDescription, Mussum Ipsum, cacilds vidis litro'
							},
							{
								type: 'mrkdwn',
								text: '*Text 2*\nDescription, Mussum Ipsum, cacilds vidis litro'
							}
						]
					},
					{
						type: 'section',
						fields: [
							{
								type: 'mrkdwn',
								text: '*Text 3*\nDescription, Mussum Ipsum, cacilds vidis litro'
							},
							{
								type: 'mrkdwn',
								text: '*Text 4*\nDescription, Mussum Ipsum, cacilds vidis litro'
							}
						]
					}
				])
			}

			<Separator title='Modal - Section Accessories' />
			{
				UiKitModal([
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Bruno Quadros*,\nPlease review your details for your *travel expense*.\nExpense no. *DA921*.'
						},
						accessory: {
							type: 'image',
							imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png'
						}
					},
					{
						type: 'divider'
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Date:*\n11/02/2020'
						}
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Category:*\nTravel'
						}
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Cost:*\n$150.00 USD'
						}
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Notes:*\nWebSummit Conference'
						}
					}
				])
			}

			<Separator title='Modal - Form Input' />
			{
				UiKitModal([
					{
						type: 'input',
						element: {
							type: 'plain_text_input'
						},
						label: {
							type: 'plain_text',
							text: 'Outgoing Title',
							emoji: true
						},
						hint: {
							type: 'plain_text',
							text: 'Pick something unique!',
							emoji: true
						}
					},
					{
						type: 'input',
						element: {
							type: 'datepicker',
							initial_date: '1990-04-28',
							placeholder: {
								type: 'plain_text',
								text: 'Select a date',
								emoji: true
							}
						},
						label: {
							type: 'plain_text',
							text: 'Set a date',
							emoji: true
						}
					},
					{
						type: 'input',
						element: {
							type: 'multi_static_select',
							options: [{
								text: {
									type: 'plain_text',
									text: 'John'
								},
								value: 1
							}, {
								text: {
									type: 'plain_text',
									text: 'Dog'
								},
								value: 2
							}]
						},
						label: {
							type: 'plain_text',
							text: 'Share with...',
							emoji: true
						}
					}
				])
			}

			<Separator title='Modal - Form TextArea' />
			{
				UiKitModal([
					{
						type: 'context',
						elements: [{
							type: 'mrkdwn',
							text: 'Task: ZOL-994'
						}]
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'Update Spec final assets'
						},
						accessory: {
							type: 'button',
							text: {
								type: 'plain_text',
								text: 'Change'
							}
						}
					},
					{
						type: 'divider'
					},
					{
						type: 'input',
						element: {
							type: 'plain_text_input',
							multiline: true
						},
						placeholder: {
							type: 'plain_text',
							text: 'Write Something',
							emoji: true
						},
						label: {
							type: 'plain_text',
							text: 'Notes',
							emoji: true
						},
						hint: {
							type: 'plain_text',
							text: 'Please take the time to compose something short',
							emoji: true
						},
						description: {
							type: 'plain_text',
							text: 'Describe your update',
							emoji: true
						}
					}
				])
			}

			<Separator title='Modal - Images' />
			{
				UiKitModal([
					{
						type: 'image',
						title: {
							type: 'plain_text',
							text: 'Example Image',
							emoji: true
						},
						imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png',
						alt_text: 'Example Image'
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'How could be the life in Mars?'
						}
					},
					{
						type: 'context',
						elements: [
							{
								type: 'image',
								imageUrl: 'https://raw.githubusercontent.com/RocketChat/Rocket.Chat.Artwork/master/Logos/icon-circle-256.png'
							},
							{
								type: 'mrkdwn',
								text: 'November 25, 2019'
							}
						]
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: '*Next stop, Mars!*\nMussum Ipsum, cacilds vidis litro abertis. Admodum accumsan disputationi eu sit. Vide electram sadipscing et per. Diuretics paradis num copo Ã© motivis de denguis. Mais vale um bebadis conhecidiss, que um alcoolatra anonimis. Aenean aliquam molestie leo, vitae iaculis nisl.'
						}
					}
				])
			}

			<Separator title='Modal - Actions' />
			{
				UiKitModal([{
					type: 'input',
					element: {
						type: 'plain_text_input'
					},
					label: {
						type: 'plain_text',
						text: 'Title',
						emoji: true
					}
				},
				{
					type: 'section',
					text: {
						type: 'mrkdwn',
						text: 'Details'
					}
				},
				{
					type: 'section',
					accessory: {
						type: 'static_select',
						options: [
							{
								value: 1,
								text: {
									type: 'plain_text',
									text: 'TypeL Task'
								}
							}, {
								value: 2,
								text: {
									type: 'plain_text',
									text: 'second button'
								}
							}]
					}
				},
				{
					type: 'section',
					accessory: {
						type: 'static_select',
						options: [
							{
								value: 1,
								text: {
									type: 'plain_text',
									text: 'Project: Space (winter)'
								}
							}, {
								value: 2,
								text: {
									type: 'plain_text',
									text: 'second button'
								}
							}]
					}
				},
				{
					type: 'section',
					accessory: {
						type: 'static_select',
						options: [
							{
								value: 1,
								text: {
									type: 'plain_text',
									text: 'Priority (optional)'
								}
							}, {
								value: 2,
								text: {
									type: 'plain_text',
									text: 'second button'
								}
							}]
					}
				},
				{
					type: 'section',
					accessory: {
						type: 'static_select',
						options: [
							{
								value: 1,
								text: {
									type: 'plain_text',
									text: 'Assinee (optional)'
								}
							}, {
								value: 2,
								text: {
									type: 'plain_text',
									text: 'second button'
								}
							}]
					}
				},
				{
					type: 'input',
					element: {
						type: 'plain_text_input',
						multiline: true
					},
					placeholder: {
						type: 'plain_text',
						text: 'Write Something',
						emoji: true
					},
					label: {
						type: 'plain_text',
						text: 'Description',
						emoji: true
					}
				}])
			}

			<Separator title='Modal - Contexts and Dividers' />
			{
				UiKitModal([
					{
						type: 'context',
						elements: [{
							type: 'mrkdwn',
							text: 'Due today'
						}]
					},
					{
						type: 'divider'
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'Finish interface componests (3 hours)'
						},
						accessory: {
							blockId: 'overflow-1',
							type: 'overflow',
							options: [
								{
									text: {
										type: 'plain_text',
										text: 'Details',
										emoji: true
									},
									value: 'value-0'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Remove',
										emoji: true
									},
									value: 'value-1'
								}
							]
						}
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'English Class (1 hour)'
						},
						accessory: {
							blockId: 'overflow-2',
							type: 'overflow',
							options: [
								{
									text: {
										type: 'plain_text',
										text: 'Details',
										emoji: true
									},
									value: 'value-0'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Remove',
										emoji: true
									},
									value: 'value-1'
								}
							]
						}
					},
					{
						type: 'section',
						text: {
							type: 'mrkdwn',
							text: 'Send an email to John (15min)'
						},
						accessory: {
							blockId: 'overflow-3',
							type: 'overflow',
							options: [
								{
									text: {
										type: 'plain_text',
										text: 'Details',
										emoji: true
									},
									value: 'value-0'
								},
								{
									text: {
										type: 'plain_text',
										text: 'Remove',
										emoji: true
									},
									value: 'value-1'
								}
							]
						}
					}
				])
			}

			<Separator title='Modal - Input with error' />
			<KitContext.Provider value={{ ...defaultContext, errors: { 'input-test': 'error test' } }}>
				<UiKitComponent
					render={UiKitModal}
					blocks={[{
						type: 'input',
						element: {
							type: 'plain_text_input',
							actionId: 'input-test'
						},
						label: {
							type: 'plain_text',
							text: 'Label',
							emoji: true
						}
					}]}
				/>
			</KitContext.Provider>

			<Separator title='Modal - Multilne with error' />
			<KitContext.Provider value={{ ...defaultContext, errors: { 'input-test': 'error test' } }}>
				<UiKitComponent
					render={UiKitModal}
					blocks={[{
						type: 'input',
						element: {
							type: 'plain_text_input',
							multiline: true,
							actionId: 'input-test'
						},
						label: {
							type: 'plain_text',
							text: 'Label',
							emoji: true
						}
					}]}
				/>
			</KitContext.Provider>

			<Separator title='Modal - DatePicker with error' />
			<KitContext.Provider value={{ ...defaultContext, errors: { 'input-test': 'error test' } }}>
				<UiKitComponent
					render={UiKitModal}
					blocks={[{
						type: 'input',
						element: {
							type: 'datepicker',
							initial_date: '1990-04-28',
							actionId: 'input-test',
							placeholder: {
								type: 'plain_text',
								text: 'Select a date',
								emoji: true
							}
						},
						label: {
							type: 'plain_text',
							text: 'Label',
							emoji: true
						}
					}]}
				/>
			</KitContext.Provider>
		</ScrollView>
	</SafeAreaView>
);
