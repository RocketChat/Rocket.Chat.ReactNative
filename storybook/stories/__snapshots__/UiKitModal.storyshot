// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots UiKitModal Modal - Actions 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Contexts and Dividers 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - DatePicker with error 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Form Input 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Form TextArea 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Images 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Input with error 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Multilne with error 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Section Accessories 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;

exports[`Storyshots UiKitModal Modal - Section and Selects 1`] = `"{\\"type\\":\\"RCTScrollView\\",\\"props\\":{\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"},{\\"paddingHorizontal\\":16}],\\"keyboardShouldPersistTaps\\":\\"always\\"},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"RCTSafeAreaView\\",\\"props\\":{\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1,\\"backgroundColor\\":\\"#fff\\"}},\\"children\\":null}]}]}"`;
