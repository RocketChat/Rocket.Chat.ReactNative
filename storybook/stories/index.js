/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
