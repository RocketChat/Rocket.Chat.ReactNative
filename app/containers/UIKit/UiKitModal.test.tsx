import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './UiKitModal.stories';

generateSnapshots(stories);
