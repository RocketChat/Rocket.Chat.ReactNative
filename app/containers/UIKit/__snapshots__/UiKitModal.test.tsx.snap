// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: ModalActions should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalContextsDividers should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalDatePickerWithError should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalFormInput should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalFormTextArea should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalImages should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalInputWithError should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalMultiSelect should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalMultilneWithError should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalSectionAccessories should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`Story Snapshots: ModalSectionSelects should match snapshot 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;
